datasource.go

```diff
@@ -118,3 +128,75 @@ func (ds *sqldatasource) CheckHealth(ctx context.Context, req *backend.CheckHeal
 		Message: "Data source is working",
 	}, nil
 }
+
+func handleError(rw http.ResponseWriter, err error) {
+	rw.WriteHeader(http.StatusBadRequest)
+	_, err = rw.Write([]byte(err.Error()))
+	if err != nil {
+		backend.Logger.Error(err.Error())
+	}
+}
+
+type columnRequest struct {
+	Table string `json:"table"`
+}
+
+func (ds *sqldatasource) handle(resource string) func(rw http.ResponseWriter, req *http.Request) {
```

Comment:
I think this might be clearer if it were split into 3 different handler functions for `Tables`, `Schemas`, and `Columns`. It's not exactly DRY but I think it will be clearer that way.

One thing that might help is setting up the functions separately,

```go

func (ds *sqldatasource) GetColumns(...) ([]string, error) {
    // look for tables, return errors...
}

func (ds *sqldatasource) HandleGetTables(w http.ResponseWriter, r *http.Request) {
    columns, err := ds.GetColumns(...)
    if err != nil {
        handleError(rw, err) // We only have to do this once now
        return
    }
    // ...
}
```